src/AddToCampaignHandler.php

```diff
@@ -171,31 +185,70 @@ protected function getObject($id, $class)
      */
     public function Form($object)
     {
-        $inChangeSets = array_unique(ChangeSetItem::get_for_object($object)->column('ChangeSetID'));
-        $changeSets = $this->getAvailableChangeSets()->map();
+        $inChangeSets = $this->getInChangeSets($object);
+        $changeSets = $this->getAvailableChangeSets();
 
-        $campaignDropdown = DropdownField::create('Campaign', '', $changeSets);
+        // Remove campaigns that the $object is already in
+        foreach ($inChangeSets as $cs) {
+            $changeSets->remove($changeSets->find('ID', $cs->ID));
+        }
+
+        $inChangeSetsText = "";
+        if ($inChangeSets->count() > 0) {
+            $inChangeSetNames = implode(', ', $inChangeSets->column('Name'));
+            $inChageSetsLabel = _t(
+                'Campaign.AddToCampaignInChangsetLabel',
+                'Heads up, this item is already in campaign(s):'
+            );
+            $inChangeSetsText = sprintf(
+                '<div class="alert alert-info"><p><strong>%s</strong><br/> %s</p></div>',
+                $inChageSetsLabel,
+                $inChangeSetNames
+            );
+        }
+
+        $campaignDropdown = DropdownField::create(
+            'Campaign',
+            _t('Campaigns.AddToCampaignAvailableLabel', 'Available campaigns'),
+            $changeSets
+        );
         $campaignDropdown->setEmptyString(_t('Campaigns.AddToCampaignFormFieldLabel', 'Select a Campaign'));
         $campaignDropdown->addExtraClass('noborder');
         $campaignDropdown->addExtraClass('no-chosen');
-        $campaignDropdown->setDisabledItems($inChangeSets);
 
         $fields = new FieldList([
+            LiteralField::create("InChangeSetsText", $inChangeSetsText),
             $campaignDropdown,
             HiddenField::create('ID', null, $this->data['ID']),
             HiddenField::create('ClassName', null, $this->data['ClassName'])
         ]);
 
-
         $form = new Form(
             $this->controller,
             $this->name,
             $fields,
             new FieldList(
                 $action = AddToCampaignHandler_FormAction::create()
+                    ->setTitle(_t('Campaigns.AddToCampaignAddAction', 'Add'))
             )
         );
 
+        // If there's no campaigns to add to, we shouldn't have the dropdown
+        // and add button
+        if ($changeSets->count() === 0) {
+            $campaignAdminUrl = CampaignAdmin::singleton()->config()->get('url_segment');
+            $link = Controller::join_links('admin', $campaignAdminUrl);
+            $noCampaingsText = _t(
+                'Campaigns.AddToCampaignNoCampaignsToAdd',
+                '<p class="lead text-xs-center">This item has been added to all available campaigns.<br/><a href="{campaignSectionLink}" class="add-to-campaign-modal__nav-link">Create a new campaign?</a></p>',
```

Comment:
This is only true if this item is in a campaign. If there are no campaigns at all this will still appear, incorrectly.